readme.md

```diff
@@ -0,0 +1,149 @@
+## 🔬 Project Overview
+
+This project is a Book Store application built using Java and Spring Boot. It is designed to manage a catalog of books, provide user authentication and authorization, and ensure secure access to various operations. The project supports CRUD operations for book management and includes a user-friendly interface for accessing the API documentation through Swagger UI.
+
+* * *
+
+## 🛠️ Technologies and Tools
+
+- **Spring Boot**: Core framework for building the application.
+- **Spring Data JPA**: Manages database interactions.
+- **Spring Security**: Handles authentication and authorization.
+- **Liquibase**: Manages database migrations.
+- **Maven**: Project management and dependency resolution.
+- **H2 Database**: In-memory database for testing (CI).
+- **MySQL**: Main database for the application.
+- **Swagger**: API documentation and testing tool.
+- **JUnit & Mockito**: For unit testing and mocking.
+- **GitHub Actions**: Continuous Integration (CI) tool.
+- **Checkstyle**: Ensures code style consistency.
+
+* * *
+
+## 💡Features
+
+- User Registration and Authentication
+- Book Browsing and Searching
+- Shopping Cart Management
+- Order Processing
+- User Profile Management
+- Admin Dashboard for Managing Books and Orders
+- Secure RESTful API Endpoints
+
+* * *
+
+## 🏗️️ Installation
+
+1. Clone the repository:
+    ```bash
+    git clone https://github.com/fedaktaras/book-store-app
+    cd online-bookstore
+    ```
+2. Configure the database: Create a new MySQL database:
+    ```sql
+    CREATE DATABASE online_book_store;
+    ```
+   Update the `application.properties` file with your MySQL database credentials:
+    ```properties
+    spring.datasource.url=jdbc:mysql://localhost:3306/online_book_store
+    spring.datasource.username=root
+    spring.datasource.password=yourpassword
+    spring.jpa.hibernate.ddl-auto=update
+    spring.jpa.show-sql=true
+    ```
+3. Install dependencies and build the project:
+    ```bash
+    mvn clean install
+    ```
+4. Run the application:
+    ```bash
+    mvn spring-boot:run
+    ```
+   The server will start on `http://localhost:8080`.
+
+### Using Docker
+
+1. Build the Docker image:
+    ```bash
+    docker-compose up
+    ```
+
+* * *
+
+## 🧑‍💻Usage
+
+- The app will be installed with a set of data.
+- Use Postman or Swagger UI to interact with the API endpoints.
+- Refer to the valid JSON files in the `try-me.txt` file (just copy/paste text to the proper endpoint).
+### API Documentation
+
+Explore the detailed API documentation generated by Swagger to understand the various endpoints and their functionalities. You can access it at `http://localhost:8080/api/swagger-ui/index.html`.
+
+###  Check out demonstration video
+In this video you can explore general overview and functionality
+`https://www.youtube.com/`
+
+* * *
+
+## 🚀Functionalities Overview
+
+### User Endpoints
+
+- Register a new user: `POST /api/auth/register`
+- Authenticate a user: `POST /api/auth/login`
```

Comment:
let's tell a bit more about the auth in your app and how does it work, where to get the token and how to use it. Maybe request/response examples for these endpoints. And add the credentials for admin and simple users because we want to give the visitor/reviewer of your project an ability to tests admin endpoints as well

Reply:
I will have preset POSTMAN to login as user and login as admin with saving token to variable. Main focus on postman. I don't plan to upload this AWS. Another my project will be there.

Reply:
then tell her that there are details and credentials for those users in those requests of the Postman collection